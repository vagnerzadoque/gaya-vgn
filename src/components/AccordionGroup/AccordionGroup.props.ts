import React from 'react';

export interface AccordionGroupProps {
  /**
   * Specify an optional className to be added to your Accordion
   */
  className?: string

  /**
   * Optional ID for testing
   */
  testID?: string;

  /**
   * Children elements
   */
  children?: React.ReactNode;
}
