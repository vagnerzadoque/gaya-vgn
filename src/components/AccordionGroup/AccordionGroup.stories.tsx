import React, { useState } from 'react'
import type { Meta, StoryObj } from '@storybook/react'
import { fn } from '@storybook/test'
import AccordionItem from '../AccordionItem'
import StoryContainer from '../../helpers/StoryContainer'
import { AccordionGroup, AccordionGroupProps } from '.'

const componentStatus = `
> AccordionGroup allows groups of several accordion components, allowing collective management and control of multiple expandable panels..

## Properties
| Property         | Values             |    Status           |
|---               |                 ---|                  ---|
| **variant**      | standard           | âœ… Available        |


## Technical Usages Examples
`

const Item = () => {
  const [isActive, setIsActive] = useState(false)

  const onClick = () => setIsActive(!isActive)

  return (
    <AccordionItem title="Lorem ipsum" isActive={isActive} onClick={onClick}>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Modi deserunt,
      ratione alias recusandae error fuga vero obcaecati impedit in dicta esse
      ad debitis iure voluptatem ipsam atque! Quam, doloribus alias!
    </AccordionItem>
  )
}

const meta: Meta<AccordionGroupProps> = {
  title: 'Components/AccordionGroup',
  component: AccordionGroup,
  parameters: {
    componentSubtitle: '',
    docs: { description: { component: componentStatus } }
  }
};

export default meta;
type Story = StoryObj<AccordionGroupProps>;

export const Playground: Story = {
  render: (args) => (
    <StoryContainer>
      <AccordionGroup {...args}>
        <Item />
        <Item />
        <Item />
      </AccordionGroup>
    </StoryContainer>
  )
};
