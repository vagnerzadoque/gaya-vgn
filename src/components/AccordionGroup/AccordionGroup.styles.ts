import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'

const styles = createUseStyles((theme: Theme) => ({
  wrapper: {
    width: '100%',
    display: 'grid',
    gap: theme.spacing.small
  }
}))

export default styles
