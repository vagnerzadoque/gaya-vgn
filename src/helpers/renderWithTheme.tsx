import React from 'react'
import { JssProvider, SheetsRegistry } from 'react-jss'
import themes from '@naturacosmeticos/natds-themes'
import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

const renderWithTheme = (component: JSX.Element) => {
  const sheets = new SheetsRegistry()

  const renderTree = render(
    <JssProvider registry={sheets}>
      <ThemeProvider theme={themes.natura.light}>
        {component}
      </ThemeProvider>
    </JssProvider>
  )

  return { styles: sheets.toString(), component: renderTree }
}

export default renderWithTheme
