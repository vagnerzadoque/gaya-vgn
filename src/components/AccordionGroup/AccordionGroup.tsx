import React from 'react'
import styles from './AccordionGroup.styles'
import { AccordionGroupProps } from './AccordionGroup.props'

const AccordionGroup = React.forwardRef<HTMLDivElement, AccordionGroupProps>(
  (props, ref) => {
    const {
      className = '',
      ...rest
    } = props
    const { wrapper } = styles()

    return (
      <div
        className={`${className} ${wrapper} AccordionGroup-gaya`}
        ref={ref}
        {...rest}
      >
        {props.children}
      </div>
    )
  }
)

export default AccordionGroup
