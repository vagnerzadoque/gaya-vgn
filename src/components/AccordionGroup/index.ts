import AccordionGroup from './AccordionGroup'

export { AccordionGroup }
export { AccordionGroupProps } from './AccordionGroup.props'
export default AccordionGroup
