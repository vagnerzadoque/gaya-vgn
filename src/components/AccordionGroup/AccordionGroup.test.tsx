import React from 'react'
import AccordionGroup from '.'
import renderWithTheme from '../../helpers/renderWithTheme'

describe('AccordionGroup component', () => {
  it('should render component with info styles if color is not defined', () => {
    const { styles, component } = renderWithTheme(
      <AccordionGroup />
    )

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
