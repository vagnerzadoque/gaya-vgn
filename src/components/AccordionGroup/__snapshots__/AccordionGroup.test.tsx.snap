// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionGroup component should render component with info styles if color is not defined 1`] = `
[
  ".wrapper-0-2-1 {
  gap: 16px;
  width: 100%;
  display: grid;
}",
  <div>
    <div
      class=" wrapper-0-2-1 AccordionGroup-gaya"
    />
  </div>,
]
`;
